tests/destinations/test_path_utils.py

```diff
@@ -1,46 +1,122 @@
+    # Extra layouts
+    ("{table_name}/{curr_date}/{load_id}.{file_id}.{ext}{timestamp}", True, []),
+    ("{table_name}/{YYYY}-{MM}-{DD}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{YYYY}-{MMM}-{D}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{DD}/{HH}/{m}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{D}/{HH}/{mm}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{timestamp}/{load_id}.{file_id}.{ext}", True, []),
+    (
+        "{table_name}/{timestamp}/{type}-{vm}-{module}/{load_id}.{file_id}.{ext}",
+        True,
+        [],
+    ),
+    ("{table_name}/dayofweek-{ddd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{ddd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{Q}/{MM}/{ddd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{YY}-{MM}-{D}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{Y}-{M}-{D}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{HH}/{mm}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{H}/{m}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{M}/{dddd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{M}/{ddd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{M}/{dd}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{M}/{d}/{load_id}.{file_id}.{ext}", True, []),
+    ("{table_name}/{load_package_timestamp}/{d}/{load_id}.{file_id}.{ext}", True, []),
+    # invalid layouts
+    ("{illegal_placeholder}{table_name}", False, ["illegal_placeholder"]),
+    (
+        "{table_name}/{abc}/{load_id}.{ext}{timestamp}-{random}",
+        False,
+        ["abc", "random"],
+    ),
+)
 
 
-def test_create_path() -> None:
-    path_vars = {
-        "schema_name": "schema_name",
-        "table_name": "table_name",
-        "load_id": "load_id",
-        "file_id": "file_id",
-        "ext": "ext",
-    }
-    path = path_utils.create_path(
-        "{schema_name}/{table_name}/{load_id}.{file_id}.{ext}", **path_vars
+@pytest.fixture
+def test_load(load_storage: LoadStorage) -> TestLoad:
+    load_id, filename = start_loading_file(load_storage, "test file")  # type: ignore[arg-type]
+    info = ParsedLoadJobFileName.parse(filename)
+    return load_id, info
+
+
+@pytest.mark.parametrize("layout,is_valid,invalid_placeholders", ALL_LAYOUTS)
+def test_layout_validity(
+    layout: str, is_valid: bool, invalid_placeholders: List[str], test_load: TestLoad
+) -> None:
+    load_id, job_info = test_load
+    if is_valid:
+        now_timestamp = pendulum.now().to_iso8601_string()
+        check_layout(layout, EXTRA_PLACEHOLDERS)
+        path = create_path(
+            layout,
+            schema_name="schema_name",
+            load_id=load_id,
+            load_package_timestamp=now_timestamp,
+            file_name=job_info.file_name(),
+            extra_placeholders=EXTRA_PLACEHOLDERS,
+        )
+        assert len(path.split("/")) == len(layout.split("/"))
```

Comment:
good sanity check but I'd expect that you put the resolved paths in the fixtures. there's nothing wrong with running the test, printing the paths, checking manually and moving them to the fixtures.

at least the reviewer can also check them. now I cannot really tell if what you do works